table/iterator.go

```diff
@@ -236,13 +236,18 @@ func (itr *Iterator) seekFromOffset(blockIdx int, offset int, key []byte) {
 }
 
 func (itr *Iterator) seekBlock(key []byte) int {
+	var baseKeyBuf []byte
 	return sort.Search(len(itr.t.blockEndOffsets), func(idx int) bool {
 		baseKeyStartOff := 0
 		if idx > 0 {
 			baseKeyStartOff = int(itr.t.baseKeysEndOffs[idx-1])
 		}
 		baseKeyEndOff := itr.t.baseKeysEndOffs[idx]
 		baseKey := itr.t.baseKeys[baseKeyStartOff:baseKeyEndOff]
+		if itr.bi.globalTs != maxGlobalTs {
+			baseKey = append(baseKeyBuf[:0], baseKey...)
+			baseKey = append(baseKey, itr.bi.globalTs[:]...)
```

Comment:
This has extra memory allocation.
We can compare baseKey and parsed key, if equal, compare ts.